front/src/applications/operationalStudies/components/Scenario/ScenarioContent.tsx

```diff
@@ -61,62 +60,41 @@ const ScenarioContent = ({
     removeTrains,
   } = useScenarioData(scenario, timetable, infra);
 
+  const [ngeDto, setNgeDto] = useState<NetzgrafikDto>();
+
+  const dtoImport = async () => {
+    const dto = await importTimetableToNGE(scenario.infra_id, scenario.timetable_id, dispatch);
+    setNgeDto(dto);
+  };
+
   const toggleMicroMacroButton = useCallback(
     (isMacroMode: boolean) => {
       setIsMacro(isMacroMode);
       if (!isMacroMode && collapsedTimetable) {
         setCollapsedTimetable(false);
       }
+      if (isMacroMode) {
+        dtoImport();
```

Comment:
Since this is in a `useEffect()` with `collapsedTimetable` as a dependency, the NGE import is performed each time the sidebar is collapsed or expanded, even if nothing has changed. I think we should keep the `dtoImport()` call in a `useEffect()` which doesn't have this dependency.